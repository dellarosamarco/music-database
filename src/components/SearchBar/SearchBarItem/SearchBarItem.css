.search-bar__result {
    display: flex;
    align-items: start;
    gap: var(--spacing-1);
    cursor: pointer;
    padding: var(--spacing-1);
    border-radius: var(--border-radius-2);
}

.search-bar__result:hover {
    background-color: var(--tertiary-color);
}

.search-bar__result-image {
    width: var(--icon-size-lg);
    border-radius: var(--border-radius-2);
}

.search-bar__result-info {
    width: 100%;
    overflow: hidden;
}