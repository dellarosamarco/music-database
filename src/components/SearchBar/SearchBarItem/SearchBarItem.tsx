import TextComponent from '../../Text/Text';
import './SearchBarItem.css';

type SearchBarItemProps = {
    name: string;
    description: string;
    image: string;
}

export const SearchBarItem = ({
    name,
    description,
    image
}: SearchBarItemProps) => {
    return (
        <div className='search-bar__result'>
            <img src={image} alt='album-icon' className='search-bar__result-image'/>
            <div className='search-bar__result-info'>
                <TextComponent>{name}</TextComponent>
                <TextComponent>{description}</TextComponent>
            </div>
        </div>
    );
}