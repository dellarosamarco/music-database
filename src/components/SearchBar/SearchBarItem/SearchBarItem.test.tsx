import { render, screen, fireEvent } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import { SearchBarItem } from './SearchBarItem';

describe('SearchBarItem Component', () => {
  const mockOnClick = vi.fn();
  const mockProps = {
    name: 'Test Album',
    description: 'Test Artist',
    image: 'https://example.com/test-album.jpg',
    onClick: mockOnClick,
  };

  it('renders the SearchBarItem with correct data', () => {
    render(<SearchBarItem {...mockProps} />);

    expect(screen.getByText('Test Album')).toBeInTheDocument();
    expect(screen.getByText('Test Artist')).toBeInTheDocument();

    const image = screen.getByAltText('album-icon');
    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute('src', mockProps.image);
  });

  it('calls onClick when the item is clicked', () => {
    render(<SearchBarItem {...mockProps} />);

    const searchBarItem = screen.getByText('Test Album').closest('.search-bar__result');
    fireEvent.click(searchBarItem!);

    expect(mockOnClick).toHaveBeenCalled();
  });

  it('matches the snapshot', () => {
    const { container } = render(<SearchBarItem {...mockProps} />);
    
    expect(container).toMatchSnapshot();
  });
});
