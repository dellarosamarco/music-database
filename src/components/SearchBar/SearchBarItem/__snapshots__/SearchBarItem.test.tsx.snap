// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchBarItem Component > matches the snapshot 1`] = `
<div>
  <div
    class="search-bar__result"
  >
    <img
      alt="album-icon"
      class="search-bar__result-image"
      src="https://example.com/test-album.jpg"
    />
    <div
      class="search-bar__result-info"
    >
      <p
        class="text-component text-component--ellipsis"
      >
        Test Album
      </p>
      <p
        class="text-component text-component--ellipsis"
      >
        Test Artist
      </p>
    </div>
  </div>
</div>
`;
